sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -946,6 +947,38 @@ object functions {
   def cosh(columnName: String): Column = cosh(Column(columnName))
 
   /**
+   * Adds a number of days to startDate, given values for startDate and days
+   *
+   * @group datetime_funcs
+   * @since 1.5.0
+   */
+  def date_add(startDate: Column, days: Column): Column = DateAdd(startDate.expr, days.expr)
+
+  /**
+   * Adds a number of days to startDate, given column names for startDate and days
+   *
+   * @group datetime_funcs
+   * @since 1.5.0
+   */
+  def date_add(startDate: String, days: String): Column = date_add(Column(startDate), Column(days))
```

Comment:
I think we probably need to think about what we need to provide for DF api. Otherwise the similar will happens again. E.g. Should we also provide the version like:
```
def date_add(startDate: Date, days: String): Column
def date_add(startDate: Timestamp, days: String): Column
...
```